src/Cassandra.IntegrationTests/Policies/Tests/RetryPolicyTests.cs

```diff
@@ -226,7 +226,114 @@ private void RetryPolicyTest(ITestCluster testCluster)
+            var testCluster = TestClusterManager.CreateNew(2, options);
+            testCluster.Builder = Cluster.Builder()
+                                         .AddContactPoint(testCluster.ClusterIpPrefix + "1");
+            testCluster.InitClient();
+
+            var tableName = TestUtils.GetUniqueTableName();
+            testCluster.Session.DeleteKeyspaceIfExists(keyspace);
+            testCluster.Session.Execute(string.Format(TestUtils.CreateKeyspaceSimpleFormat, keyspace, 2), ConsistencyLevel.All);
+            testCluster.Session.ChangeKeyspace(keyspace);
+            testCluster.Session.Execute(new SimpleStatement(string.Format("CREATE TABLE {0} (k int PRIMARY KEY, i int)", tableName)).SetConsistencyLevel(ConsistencyLevel.All));
 
+            //testCluster.PauseNode(2);
 
+            var testPolicy = new TestRetryPolicy();
+            var policy = new IdempotenceAwareRetryPolicy(testPolicy);
+
+            try
+            {
+                testCluster.Session.Execute(new SimpleStatement(string.Format("INSERT INTO {0}(k, i) VALUES (0, 0)", tableName))
+                    .SetIdempotence(true)
+                    .SetConsistencyLevel(ConsistencyLevel.All)
+                    .SetRetryPolicy(policy));
+            }
+            catch (Exception exception)
+            {
+                if (exception is WriteTimeoutException)
+                {
+                    //throws a WriteTimeoutException, as its set as an idempotent query, it will call the childPolicy
+                    Assert.AreEqual(0, testPolicy.ReadTimeoutCounter);
+                    Assert.AreEqual(1, testPolicy.WriteTimeoutCounter);
+                    Assert.AreEqual(0, testPolicy.UnavailableCounter);
+                }
+                else
+                {
+                    //ignore, sometimes node1 will be aware the the node2 was paused and it will throw an UnavailableException
+                }
+                
+            }
+
+            //testing with unidempotent query
+            testPolicy.UnavailableCounter = 0;
+            
+            try
+            {
+                testCluster.Session.Execute(new SimpleStatement(string.Format("INSERT INTO {0}(k, i) VALUES (0, 0)", tableName))
+                    .SetIdempotence(false)
+                    .SetConsistencyLevel(ConsistencyLevel.All)
+                    .SetRetryPolicy(policy));
+            }
+            catch (Exception exception)
+            {
+                if (exception is WriteTimeoutException)
+                {
+                    //throws a WriteTimeoutException, as its set as an idempotent query, it will call the childPolicy
+                    Assert.AreEqual(0, testPolicy.ReadTimeoutCounter);
+                    Assert.AreEqual(0, testPolicy.WriteTimeoutCounter);
+                    Assert.AreEqual(0, testPolicy.UnavailableCounter);
+                }
+                else
```

Comment:
Validate that is either `WriteTimeoutException` or `UnavailableException`.
You can do it by using 2 different catch blocks or `Assert.That(ex, Is.InstanceOf<WriteTimeoutException>().Or.InstanceOf<UnavailableException>())`